mmv1/third_party/terraform/tests/data_source_dns_record_set_test.go.erb

```diff
@@ -0,0 +1,103 @@
+<% autogen_exception -%>
+package google
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccDataSourceDnsRecordSet_basic(t *testing.T) {
+	t.Parallel()
+
+	var ttl1, ttl2 string // ttl is a computed string-type attribute that is easy to compare in the test
+
+	managedZoneName := fmt.Sprintf("tf-test-zone-%s", randString(t, 10))
+
+	vcrTest(t, resource.TestCase{
+		PreCheck: func() { testAccPreCheck(t) },
+		CheckDestroy: testAccCheckDnsRecordSetDestroyProducerFramework(t),
+		Steps: []resource.TestStep{
+			{
+				ExternalProviders: providerVersion450(),
+				Config:            testAccDataSourceDnsRecordSet_basic(managedZoneName, randString(t, 10), randString(t, 10)),
+				Check: resource.ComposeTestCheckFunc(
+					checkDataSourceStateMatchesResourceState("data.google_dns_record_set.rs", "google_dns_record_set.rs"),
+					testExtractResourceAttr("data.google_dns_record_set.rs", "ttl", &ttl1),
+				),
+			},
+			{
+				ProtoV5ProviderFactories: protoV5ProviderFactories(t),
+				Config:                   testAccDataSourceDnsRecordSet_basic(managedZoneName, randString(t, 10), randString(t, 10)),
+				Check: resource.ComposeTestCheckFunc(
+					checkDataSourceStateMatchesResourceState("data.google_dns_record_set.rs", "google_dns_record_set.rs"),
+					testExtractResourceAttr("data.google_dns_record_set.rs", "ttl", &ttl2),
+					testCheckAttributeValuesEqual(&ttl1, &ttl2),
+				),
+			},
+		},
+	})
+}
+
+func testAccDataSourceDnsRecordSet_basic(managedZoneName, zoneName, recordSetName string) string {
+	return fmt.Sprintf(`
+resource "google_dns_managed_zone" "zone" {
+  name     = "%s"
+  dns_name = "%s.hashicorptest.com."
+}
+
+resource "google_dns_record_set" "rs" {
+  managed_zone = google_dns_managed_zone.zone.name
+  name         = "%s.${google_dns_managed_zone.zone.dns_name}"
+  type         = "A"
+  ttl          = 300
+  rrdatas      = [
+	"192.168.1.0",
+  ]
+}
+
+data "google_dns_record_set" "rs" {
+  managed_zone = google_dns_record_set.rs.managed_zone
+  name         = google_dns_record_set.rs.name
+  type         = google_dns_record_set.rs.type
+}
+`, managedZoneName, zoneName, recordSetName)
+}
+
+// Framework checkdestroy
```

Comment:
It'd be useful to make this comment similar to the one above the other function for managed zones

```suggestion
// testAccCheckDnsRecordSetDestroyProducerFramework is the framework version of the generated testAccCheckDnsRecordSetDestroyProducer
// when we automate this, we'll use the automated version and can get rid of this
```